File: output/CodeLlama/avatar/Java/Python/codeforces_30_A.py
import sys

A, B, n = map(int, sys.stdin.readline().split())

if A == 0:

    if B == 0:

        print(1)

    else:

        print("No solution")

    return

if B % A != 0:

    print("No solution")

    return

B /= A

neg = B < 0

if neg and n % 2 == 0:

    print("No solution")

    return

if neg:

    B = -B

for x in range(B + 1):

    if pow(x, n) == B:

        print(x if not neg else -x)

        return

print("No solution")

File: output/CodeLlama/avatar/Python/Java/codeforces_30_A.java
public class codeforces_30_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int A = sc.nextInt();
        int B = sc.nextInt();
        int n = sc.nextInt();
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                System.out.println(X);
                return;
            }
        }
        System.out.println("No solution");
    }
}